app/objective/get.controller.js

```diff
@@ -1,26 +1,78 @@
 const { logger } = require('../../common/logging/logger')
-const { getSentencePlanObjective } = require('../../common/data/sentencePlanningApi')
-const { removeUrlLevels } = require('../../common/utils/util')
+const { getSentencePlanObjective, getSentencePlanNeeds } = require('../../common/data/sentencePlanningApi')
+const { removeUrlLevels, sortObject } = require('../../common/utils/util')
 
 const getObjective = async (
-  { path, errors, errorSummary, body, renderInfo, params: { planId, objectiveId }, session: { 'x-auth-token': token } },
+  {
+    path,
+    errors,
+    errorSummary,
+    body: { objective = null, needs = [] },
+    renderInfo,
+    params: { planId, objectiveId },
+    session: { 'x-auth-token': token },
+  },
   res
 ) => {
   const nexturl = path.substring(0, path.lastIndexOf('/'))
   const backurl = removeUrlLevels(path, 2)
   const renderDetails = { ...renderInfo, nexturl, backurl }
+  let displayObjective = { description: '', needs: [] }
 
+  // get the saved or submitted objective
   try {
-    if (body.objective !== undefined) {
-      renderDetails.objective = body.objective
+    if (objective) {
+      displayObjective.description = objective
+      displayObjective.needs = needs || []
     } else if (objectiveId.toLowerCase() !== 'new') {
-      renderDetails.objective = await getSentencePlanObjective(planId, objectiveId, token)
+      const savedObjective = await getSentencePlanObjective(planId, objectiveId, token)
+
+      // convert need to an array if only one has been passed back
+      if (savedObjective.needs) {
+        if (!(savedObjective.needs instanceof Array)) {
```

Comment:
I'd argue utilise a single `if` with an  `and`.
` if (savedObjective.needs && !(savedObjective.needs instanceof Array)) {`